metaflow/plugins/argo/argo_workflows.py

```diff
@@ -1282,6 +1363,12 @@ def _visit(
                                         "{{tasks.%s.outputs.parameters.split-cardinality}}"
                                         % self._sanitize(node.name)
                                     ),
+                                    # Only pick the output step from the first iteration of the foreach task, as it should be identical for all.
```

Comment:
this assertion is untrue. the conditional step inside a foreach can vary based on each executing task.

as such, the current deterministic task-ids that we have for foreach tasks will not work when a conditional branch is the step being joined, e.g. the last step in a foreach

Reply:
this type of DAG structure has been blocked for the feature until we have a proper fix for input-paths. #2551 is an attempt at this